### PR TITLE
std::chrono::system_clock and std::chrono::steady_clock might have different duration types.

### PR DESCRIPTION
**Pull request by toregge**
@baldersheim : please review

### CODE REVIEW

**Comment by baldersheim** [comment]
Where is this a problem ?

**Comment by baldersheim** [comment]
I thought that duration was independent of the clocks.

**Comment by toregge** [comment]
On Darwin (using llvm from brew), system_clock has a duration using microseconds while steady_clock has a duration using nanoseconds.

File: config/src/tests/trace/trace.cpp [hunk 1]
```diff
@@ -10,7 +10,7 @@ using namespace vespalib::slime;
 
 struct FixedClock : public Clock
 {
-    FixedClock() : _currentTime(duration::zero()) { }
+    FixedClock() : _currentTime(vespalib::system_time::duration::zero()) { }
```
[the remark this answers is not in the document]
Comment (toregge) [reply config/src/tests/trace/trace.cpp]: Fixed

File: config/src/tests/trace/trace.cpp [hunk 2]
```diff
@@ -57,7 +57,7 @@ TEST("that trace can be copied") {
     EXPECT_EQUAL(trace.toString(), trace2.toString());
 }
 
-constexpr vespalib::system_time epoch(duration::zero());
+constexpr vespalib::system_time epoch(vespalib::system_time::duration::zero());
```
[the remark this answers is not in the document]
Comment (toregge) [reply config/src/tests/trace/trace.cpp]: Fixed

File: vespalib/src/tests/trace/trace_serialization.cpp [hunk 3]
```diff
@@ -21,7 +21,7 @@ TEST("that a single trace node is serialized") {
     EXPECT_FALSE(i["payload"].valid());
 }
 
-constexpr system_time zero_system_time(duration::zero());
+constexpr system_time zero_system_time(system_time::duration::zero());
```
[the remark this answers is not in the document]
Comment (toregge) [reply vespalib/src/tests/trace/trace_serialization.cpp]: Fixed

File: vespalib/src/tests/trace/trace.cpp [hunk 4]
```diff
@@ -334,7 +334,7 @@ TEST("testVisiting")
     EXPECT_EQUAL(encoder.str, b1.encode());
 }
 
-constexpr system_time zero(duration::zero());
+constexpr system_time zero(system_time::duration::zero());
```
[the remark this answers is not in the document]
Comment (toregge) [reply vespalib/src/tests/trace/trace.cpp]: Fixed

File: storage/src/tests/visiting/memory_bounded_trace_test.cpp [hunk 5]
```diff
@@ -7,7 +7,7 @@ using namespace ::testing;
 
 namespace storage {
 
-constexpr vespalib::system_time epoch(vespalib::duration::zero());
+constexpr vespalib::system_time epoch(vespalib::system_time::duration::zero());
```
[the remark this answers is not in the document]
Comment (toregge) [reply storage/src/tests/visiting/memory_bounded_trace_test.cpp]: Fixed

**Pull request merged**